### PR TITLE
add CLANG_TIDY_CONFIG_FILE

### PR DESCRIPTION
**Pull request by zingale**
this allows us to use the AMReX GNU Make clang-tidy build

depends on https://github.com/AMReX-Codes/amrex/pull/3202

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
okay, docs are updated and this is ready for review

File: Source/driver/Castro.H [hunk 1]
```diff
@@ -702,7 +702,7 @@ public:
 /// @param n_error_buf  integer
 /// @param ngrow        Number of ghost cells
 ///
-    void errorEst (amrex::TagBoxArray& tb,
+    void errorEst (amrex::TagBoxArray& tags,
```
Comment (yut23) [inline Source/driver/Castro.H:705]: doxygen comment should be updated

File: Source/sources/Castro_sources.H [hunk 2]
```diff
@@ -268,7 +279,7 @@
 /// @param S_new    New state
 /// @param ext_src  MultiFab to fill with sources
 ///
-    void fill_thermo_source(amrex::MultiFab& S_new, amrex::MultiFab& ext_src);
+    void fill_thermo_source(amrex::MultiFab& state_in, amrex::MultiFab& thermo_src);
```
Comment (yut23) [inline Source/sources/Castro_sources.H:282]: doxygen comments should be updated

File: Source/driver/Castro.H (hunk 1 above)
Comment (zingale) [reply Source/driver/Castro.H:705]: done

**Review by yut23 — APPROVED** [review]

**Pull request merged**